src/config-file.c

```diff
@@ -76,38 +80,64 @@ static gboolean get_key_bool(GKeyFile *key_file, const gchar* group, const gchar
                 *value = TRUE;
                 return TRUE;
         }
+
+        g_set_error(error, G_KEY_FILE_ERROR,
+                    G_KEY_FILE_ERROR_INVALID_VALUE,
+                    "Value '%s' cannot be interpreted as a boolean.", val);
+
         return FALSE;
 }
 
 static gboolean get_key_int(GKeyFile *key_file, const gchar* group, const gchar* key, gint* value, const gint default_value, GError **error)
 {
-        gint val = g_key_file_get_integer(key_file, group, key, NULL);
-        if (val == 0) {
+        GError *ierror = NULL;
+        gint val = g_key_file_get_integer(key_file, group, key, &ierror);
+
+        if (val == 0 && g_error_matches(ierror, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_KEY_NOT_FOUND)) {
+                g_clear_error(&ierror);
                 *value = default_value;
                 return TRUE;
         }
+        else if (val == 0 && ierror) {
+                g_propagate_error(error, ierror);
+                return FALSE;
+        }
         *value = val;
         return TRUE;
 }
 
-static gboolean get_group(GKeyFile *key_file, const gchar *group, GHashTable **hash, GError *error)
+static gboolean get_group(GKeyFile *key_file, const gchar *group, GHashTable **hash, GError **error)
 {
         guint key;
         gsize num_keys;
-        gchar **keys, *value;
+        gchar **keys;
 
         *hash = g_hash_table_new(g_str_hash, g_str_equal);
-        keys = g_key_file_get_keys(key_file, group, &num_keys, &error);
+        keys = g_key_file_get_keys(key_file, group, &num_keys, error);
+        if (keys == NULL)
+                return FALSE;
+
+        if (num_keys == 0) {
+                g_set_error(error, G_KEY_FILE_ERROR,
+                            G_KEY_FILE_ERROR_PARSE,
+                            "Group '%s' has no keys set", group);
+                return FALSE;
+        }
+
         for (key = 0; key < num_keys; key++)
         {
-                value = g_key_file_get_value(key_file,
-                                             group,
-                                             keys[key],
-                                             &error);
+                gchar *value = g_key_file_get_value(key_file,
+                                                    group,
+                                                    keys[key],
+                                                    error);
+                if (value == NULL)
+                        return FALSE;
```

Comment:
Do we need a g_set_error here (above return) since we return FALSE?

Reply:
same as above